common/lc_model.cpp

```diff
@@ -1266,10 +1266,10 @@ void lcModel::GetScene(lcScene& Scene, lcCamera* ViewCamera, bool DrawInterface,
 	Scene.End();
 }
 
-void lcModel::AddSubModelRenderMeshes(lcScene& Scene, const lcMatrix44& WorldMatrix, int DefaultColorIndex, lcRenderMeshState RenderMeshState, bool ParentActive) const
+void lcModel::AddSubModelRenderMeshes(lcScene& Scene, const lcMatrix44& WorldMatrix, int DefaultColorIndex, lcRenderMeshState RenderMeshState, bool ParentActive, bool Preview) const
 {
 	for (lcPiece* Piece : mPieces)
-		if (Piece->GetStepHide() == LC_STEP_MAX)
+		if ((Piece->GetStepHide() == LC_STEP_MAX && Preview) || Piece->IsVisible(mCurrentStep))
```

Comment:
I don't think this check is correct, this should probably be "if (Piece->GetStepHide() == LC_STEP_MAX && !Piece->IsHidden())".

Inserted submodels are always drawn at LC_STEP_MAX, so the IsVisible check can be wrong (maybe mCurrentStep should be in the View class for a correct Document/View design).

What makes the preview drawing different to require the extra parameter?

Reply:
The extra parameter is required, because otherwise the Piece is also hidden in the parts selector widget. In this example I hid the rim of the wheel, but it's also not shown in the widget on the right.

![image](https://user-images.githubusercontent.com/4201214/54265959-b7958e00-4576-11e9-8f8a-c65c5202f62b.png)